stable/yugabyte/templates/hooks/setup-credentials-job.yaml

```diff
@@ -0,0 +1,80 @@
+{{- if or .Values.authCredentials.ycql.user .Values.authCredentials.ycql.password .Values.authCredentials.ycql.keyspace .Values.authCredentials.ysql.password .Values.authCredentials.ysql.user .Values.authCredentials.ysql.database }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "yugabyte.fullname" . }}-setup-credentials
+  namespace: "{{ .Release.Namespace }}"
+  labels:
+    app: "setup-credentials"
+    release: {{ .Release.Name | quote }}
+    chart: "{{ .Chart.Name }}"
+    component: "{{ .Values.Component }}"
+  annotations:
+    "helm.sh/hook": post-install
+    "helm.sh/hook-weight": "0"
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  backoffLimit: 2
+  template:
+    metadata:
+      name: "setup-credentials"
+      labels:
+        app: "setup-credentials"
+        release: {{ .Release.Name | quote }}
+        chart: "{{ .Chart.Name }}"
+        component: "{{ .Values.Component }}"
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: setup-credentials
+        image: "{{ .Values.Image.repository }}:{{ .Values.Image.tag }}"
+        env:
+        {{- if .Values.authCredentials.ysql.user }}
+        - name: YSQL_USER
+          value: "{{ .Values.authCredentials.ysql.user }}"
+        {{- end }}
+        {{- if .Values.authCredentials.ysql.password }}
+        - name: YSQL_PASSWORD
+          value: "{{ .Values.authCredentials.ysql.password }}"
+        {{- end }}
+        {{- if .Values.authCredentials.ysql.database }}
+        - name: YSQL_DB
+          value: "{{ .Values.authCredentials.ysql.database }}"
+        {{- end }}
+        {{- if .Values.authCredentials.ycql.user }}
+        - name: YCQL_USER
+          value: "{{ .Values.authCredentials.ycql.user }}"
+        {{- end }}
+        {{- if .Values.authCredentials.ycql.password }}
+        - name: YCQL_PASSWORD
+          value: "{{ .Values.authCredentials.ycql.password }}"
+        {{- end }}
+        {{- if .Values.authCredentials.ycql.keyspace }}
+        - name: YCQL_KEYSPACE
+          value: "{{ .Values.authCredentials.ycql.keyspace }}"
+        {{- end }}
+        {{- if .Values.tls.enabled }}
+        - name: SSL_CERTFILE
+          value: "/root/.yugabytedb/root.crt"
+        {{- end }}
+        command:
+        - 'bash'
+        - '/home/yugabyte/bin/setup-credentials/setup-credentials.sh'
```

Comment:
Yeah because of this only container will know the script is executed successfully or not.

Reply:
Makes sense - but it looks like when this cmd fails, the entire release is considered failed? Can you try that scenario to make sure that the error is easy to diagnose?

Reply:
In this case, the debugging is easy because, in the end, it gives the error something like this 'post-hook failed'. We're removing the post-install hook job's pod only on success so, in case of failure, we can check the logs of the job and can do further debugging.